### DISCUSSION TITLE
New argument --api-load-ui

### DISCUSSION THREAD

**Discussion by yo1nkers** [vladmandic/sdnext#1802] [Ideas]
EDIT: FOR PEOPLE LOOKING OR PERHAPS OVERLOOKING AS I DID, **_VLAD DOES NOT REQUIRE ARGUMENTS FOR THE API TO BE STARTED._**
YOU CAN USE THIS WITH YOUR BOTS ORIGINALLY FOR AUTO1111.

Just a quick suggestion to add the new argument --api-load-ui

I have created my own bot which links into the vlad auto ui to use stable diffusion, and I need to be able to see the WebUi in order to change the actual model, or it makes it quite difficult to just keep switching arguments.

**Comment by vladmandic** [comment]
I'm not sure I understand the use case?

**Comment by yo1nkers** [comment]
My apologies if I hadn't clarified well enough.

I am running a Discord Bot which uses the Vlad Auto1111 API to generate images, however it relies on the WebUI itself to change certain settings such as Clip Skip, VAE and Stable Diffusion Models. Meaning I have to be able to go to the UI on the localhost whilst the API is on.

When I launch the "webui" from batch file with --api-only --api-log arguments, and then I try to go to the localhost, I will see "{"detail":"Not Found"}". I understand this is intentional with --api-only.

What I am asking for is a way to use the API and have the WebUI open at the same time. --api-load-ui

**Comment by vladmandic** [comment]
but....isn't that the same as running without `--api-only`???
also, why do you need ui to set clip-skip, vae, model - do that via api also.

**Comment by yo1nkers** [comment]
My apologies, is that implying that the API is enabled regardless of the argument "--api-only"? My only intent is to be able to use the Vlad WebUI as an admin panel of sorts.

I understand I can set up clip skip, vae amd model to function from the API but these are options I'd specifically like to keep hidden from people who are using the bot.

Sorry if this is seeming like a waste of time. I am just looking to have the API and WebUI function at the same time.

**Reply by yo1nkers** [reply]
Well I'll be damned. I have discovered the API is automatically enabled with this fork of auto1111, whereas the original Stable Diffusion WebUI requires launch arguments in order to use the API. Additionally, this was part my own issue with my firewall blocking a specific port that the WebUI was listening out for.


Regardless thank you for making it clear that I do not have to use the `--api-only` launch argument for api.


tl;dr severely overlooked the API function. all good now.
